pandas/core/algorithms.py

```diff
@@ -1776,6 +1820,43 @@ def take_nd(
 take_1d = take_nd
 
 
+def take_1d_array(
```

Comment:
Did you read the docstring? It's a `take_1d` variant specialized for arrays, so I think the name makes some sense. But happy to hear alternatives.

The current `take_1d` is just an alias of `take_nd`, while here I want to have an actual specialized version.

I could also rename all occurrences of `take_1d` to `take_nd` (since they are aliases anyway, that shouldn't matter), so the "take_1d" name is free to use.

Reply:
I read the doc-string and I understand the code, but I think this additional is very confusing. You are reading it in the context of this PR, where its not terrible. However looking at the actual code is it is so non-obvious what is going on and appears to be duplicative. Please try to refactor.

Reply:
> Looking at the actual code is it is so non-obvious what is going on and appears to be duplicative.

And then the docstring explains why it is duplicative... Do you have a concrete suggestion?

(I already suggested getting rid of the current `take_1d` alias if that helps removing the confusion)